### ISSUE TITLE
Fatal error: primaryKey() 

### ISSUE THREAD

**Issue by tarzinio** [thecodeholic/php-mvc-framework#21]
Fatal  error: Uncaught Error: Non-static method  thecodeholic\phpmvc\db\DbModel::primaryKey() cannot be called statically  in C:\wamp64\www\link\vendor\thecodeholic\php-mvc-core\Application.php  on line 54

![obraz](https://user-images.githubusercontent.com/29274594/160383650-aa0575bc-fae4-412f-822c-5029921bc2bc.png)

**Comment by thecodeholic** [comment]
Thanks for submitting the issue.
I will do my best to fix that ASAP. If you want you can make a PR I would merge it.
The changes should be made here. 
https://github.com/thecodeholic/tc-php-mvc-core/blob/master/db/DbModel.php#L23

**Comment by HoosierExperienceDesigner** [comment]
I'm experiencing a similar issue but mine stems from the Application.php file. Everything was working perfect until about the 4:25:00 point in the lesson. I used var_dump at the bottom of Application constructor to target the specific bug, and for me it is occurring in this line: 

$primaryKey = $this->userClass::primaryKey(); 

Fatal error: Uncaught Error: Non-static method app\models\User::primaryKey() cannot be called statically in D:\xampp\htdocs\I211_Project\core\Application.php:45 Stack trace: #0 D:\xampp\htdocs\I211_Project\public\index.php(24): app\core\Application->__construct('D:\\xampp\\htdocs...', Array) #1 {main} thrown in D:\xampp\htdocs\I211_Project\core\Application.php on line 45

**Comment by RobertoVasquez01** [comment]
All these problems with static calls are due to the php version. It has to be higher than 7.4.
You can temporarily modify the function so that it is not static, but later in the lesson you will experience a similar problem. That's why it's better to **update the php version**

**Comment by tarzinio** [comment]
I am using version 8.1

pon., 4 kwi 2022 o 06:56 RobertoVasquez01 ***@***.***>
napisał(a):

**Comment by HoosierExperienceDesigner** [comment]
I also applied PHP settings to 8.0.1, and to be sure ran 'php -v' in the terminal with the following result: 

PS D:\xampp\htdocs\I211_Project> php -v
PHP 8.0.1 (cli) (built: Jan  5 2021 23:43:39) ( ZTS Visual C++ 2019 x64 )
Copyright (c) The PHP Group
Zend Engine v4.0.1, Copyright (c) Zend Technologies


I still get the same fatal error Non-static method app\models\User::primaryKey() cannot be called statically

**Comment by RobertoVasquez01** [comment]
Yes, I was wrong in the first part of my comment. It was because I'm using php version 7.3 and throughout the video tutorial I had to make a few small changes to take advantage of this great input from @thecodeholic, and I was quick to say that the error is based on the php version.
You have to MOMENTARY change the declaration: "abstract public static function primaryKey(): string;" from the DbModel class to this: "public function primaryKey(): string;" and call it like this: DbModel->primaryKey();
I repeat: This is momentary until the video tutorial continues, which then will have to return the abstract declaration... and it will no longer give you a problem.

**Comment by HoosierExperienceDesigner** [comment]
Thank you for the information! Will give this a shot and provide feedback on whether or not it was successful.

**Comment by HoosierExperienceDesigner** [comment]
I eventually started over and rebuilt the app from the beginning and got it to work.

**Comment by RobertoVasquez01** [comment]
**Perfect!**

**Comment by RobertoVasquez01** [comment]
Perfecto!
**Gracias a @thecodeholic**

**Comment by RobertoVasquez01** [comment]
Perfect!
Thanks to @thecodeholic

**Comment by RobertoVasquez01** [comment]
Declare the findOne function as static in the DbModel class.
**public static function findOne($where){
}**
